### PR TITLE
Tuning notebook updated to include csv upload

### PR DESCRIPTION
**Pull request by nidhinpd-YML**
- Code to provide csv as training dataset included.
- Separate and detailed section for Dataset preparation.
- Added more details about different types of dataset we can provide.

### CODE REVIEW

File: quickstarts/Tuning.ipynb [hunk 1]
```diff
@@ -48,7 +48,7 @@
       "source": [
```
Comment (pcoet) [inline quickstarts/Tuning.ipynb:48]: <p>I wonder if we actually want to talk about the protos. I think we've been trying to focus on the higher-level abstractions.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/300/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (pcoet) [inline quickstarts/Tuning.ipynb:48]: <p>Same comment here: Perhaps we should not document the protos.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/300/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

**Comment by pcoet** [comment]
@MarkDaoust Could you take a look at this? I had a couple of questions about documenting protos.

File: quickstarts/Tuning.ipynb (hunk 1 above)
Comment (nidhinpd-YML) [reply quickstarts/Tuning.ipynb:48]: **Protos** mentions removed

Comment (nidhinpd-YML) [reply quickstarts/Tuning.ipynb:48]: **Protos** mentions removed

**Review by pcoet — APPROVED** [review]

File: quickstarts/Tuning.ipynb (hunk 1 above)
Comment (markmcd) [inline quickstarts/Tuning.ipynb:48]: <blockquote>a Google Sheets file</blockquote><p>IIRC this is only supported for a <em>public</em> Google Sheets URL.</p><p> </p><blockquote>The tuning API will read a csv file with pd.read_csv and handle it as a Mapping.&nbsp;</blockquote><p><code>pd.read_csv</code>  is an implementation detail, and I realise we have documented it, but we should limit any further leakage.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/300/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (markmcd) [inline quickstarts/Tuning.ipynb:48]: <p>This isn't strictly true, but is the default. If users want to provide data in a format they already have, with any number of columns, they can specify <code>input_key</code> and <code>output_key</code> - <a href="https://github.com/google-gemini/generative-ai-python/blob/v0.8.2/google/generativeai/models.py#L258-L259" target="_blank">e.g.</a></p><p>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/300/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (markmcd) [inline quickstarts/Tuning.ipynb:48]: <p>We should avoid having commented-out code checked in (except in very specific cases) as we don't know if it stops working.</p><p> </p><p>Un-comment everything here, and isolate the branching point in a single clear line.</p><p> </p><p>e.g.</p><p> </p>

<pre class="ql-syntax" spellcheck="false">dict_data = [{...}, ...]
...
csv_file = 'data.csv'
...

# Here you can specify any of the supported formats, e.g. csv_file or dict_data
training_data = csv_file
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/300/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (markmcd) [inline quickstarts/Tuning.ipynb:48]: <pre class="ql-syntax" spellcheck="false">Line #4.    # train_dataset=pathlib.Path('/content/data.csv')
</pre><p>nit: drop the <code>/content</code> part of the path - that is only true on Colab. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/300/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

**Comment by markmcd** [comment]
> Could you take a look at this? I had a couple of questions about documenting protos.

I've done a drive-by. Agree on the protos comment - thanks for fixing that @nidhinpd-YML.

File: quickstarts/Tuning.ipynb [hunk 2]
```diff
@@ -164,23 +164,226 @@
...
+      "metadata": {
+        "id": "TD3_PjOI45Jw"
+      },
+      "source": [
+        "You can provide your CSV file to the tuning API in one of the following ways:\n",
+        "  * A path of type `str` or `pathlib.Path` to a local CSV file.\n",
+        "  * A URL to the CSV file.\n",
+        "  * The URL of a Google Sheets file.\n",
+        "\n",
+        "The tuning API will read a CSV file with `pd.read_CSV` and handle it as a `Mapping`.\n",
```
Comment (nidhinpd-YML) [inline quickstarts/Tuning.ipynb:303]: read_CSV --> read_csv

File: quickstarts/Tuning.ipynb (hunk 1 above)
Comment (nidhinpd-YML) [reply quickstarts/Tuning.ipynb:48]: Done

Comment (nidhinpd-YML) [reply quickstarts/Tuning.ipynb:48]: Done

Comment (nidhinpd-YML) [reply quickstarts/Tuning.ipynb:48]: Updated documentation to reflect this

File: quickstarts/Tuning.ipynb (hunk 2 above)
Comment (nidhinpd-YML) [reply quickstarts/Tuning.ipynb]: Removed the whole line in the new update

**Review by markmcd — APPROVED** [review]
Awesome - thanks Nidhin

**Pull request merged**